### PR TITLE
CI Docs: Using an inclusive linter

### PR DESCRIPTION
**Pull request by jeremydelacruz**
# Pull Request Template

## What are you trying to address

- Add content around using an inclusive linter in CI pipelines
- Addresses `inclusive-dev-tools` issue https://github.com/inclusive-dev-tools/woke/issues/35

## Description of new changes

- Adding new markdown file in `docs/continuous-integration/inclusive-linting.md`
- Adding content around what inclusive linting is, how it can be useful, and how to get started with an inclusive linting tool

## For all pull requests

- [x] Link to the issue you are solving (so it gets closed)
- [x] Label the pull request with the appropriate area(s)
- [x] Assign potential reviewers (you may also want to contact them on Teams to ensure timely reviews)
- [x] Assign the project - Engineering Playbook Backlog
- [x] Assign the pull request to yourself

### CODE REVIEW

File: docs/continuous-integration/inclusive-linting.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Inclusive Linting
```
Comment (madossan01) [inline docs/continuous-integration/inclusive-linting.md:1]: Love to see this guidance for CI. Thanks to propose and create the document.

File: docs/continuous-integration/inclusive-linting.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+An inclusive linter may include its own dictionary of "default" non-inclusive words and phrases to run against as a good starting point. These tools can also be customizable, oftentimes offering the ability to omit some terms and/or add your own.
+
+The ability to add additional terms to your linter has the added benefit of enabling linting of sensitive language on top of inclusive linting. This can prevent things such as customer names or other non-public information from making it into your git history, for instance.
+
+## Getting Started with an Inclusive Linter
+
+### [`woke`]
+
+One inclusive linter we recommend is `woke`. It is a language-agnostic CLI tool that detects non-inclusive language in your source code and recommends alternatives. While `woke` automatically applies a [default ruleset] with non-inclusive terms to lint for, you can also apply a custom rule config (via a yaml file) with additional terms to lint for. See [`example.yaml`] for an example of adding custom rules.
```
Comment (madossan01) [inline docs/continuous-integration/inclusive-linting.md:19]: Should we recommend or support? The questions is due to some internal approval process that may be in place first (not sure if they were already finished). @fnocera please can you advise here?

File: docs/continuous-integration/inclusive-linting.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+test.txt:2:2-11: `Blacklist` may be insensitive, use `denylist`, `blocklist` instead (warning)
+* Blacklist
+  ^
+```
+
+`woke` can be run locally on your machine or CI/CD system and is also available as a [GitHub Action]. For more information about configuration and usage, see the official [docs].
+
+[`woke`]: https://github.com/get-woke/woke
+[GitHub Action]: https://github.com/marketplace/actions/run-woke
+[docs]: https://docs.getwoke.tech/
```
[the remark this answers is not in the document]
Comment (madossan01) [reply docs/continuous-integration/inclusive-linting.md:37]: Agree. There is another PR merged yesterday that approached similar scenario for spikes. Here is the [link](https://github.com/microsoft/code-with-engineering-playbook/tree/main/docs/continuous-integration/markdown-linting) to the merged documentation for reference:

File: docs/continuous-integration/inclusive-linting.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+1. Run your pipeline
+1. View the output in the "Actions" tab in the main repository view
+
+For more information about additional configuration and usage, see the official [docs].
+
+[`woke`]: https://github.com/get-woke/woke
+[default ruleset]: https://github.com/get-woke/woke/blob/main/pkg/rule/default.yaml
+[`example.yaml`]: https://github.com/get-woke/woke/blob/main/example.yaml
+[GitHub Action]: https://github.com/marketplace/actions/run-woke
+[docs]: https://docs.getwoke.tech/
```
Comment (mkcomer) [inline docs/continuous-integration/inclusive-linting.md:65]: Add a link to GitHub Action w/ Review Dog: https://github.com/get-woke/woke-action-reviewdog ?

**Pull request merged**